docs/current/742989-cookbook.md

```diff
@@ -296,6 +375,31 @@ The following code listing exports the contents of a container directory to the
 
 [Learn more](./guides/421437-work-with-host-filesystem.md)
 
+### Publish image to registry with multiple tags
+
+The following code listing tags a container image multiple times and publishes it to a remote registry (Docker Hub). Replace the `DOCKER-HUB-USERNAME` and `DOCKER-HUB-PASSWORD` placeholders with your Docker Hub username and password respectively.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./cookbook/snippets/multiple-tags/main.go
```

Comment:
In the Go example, I would use `range` instead, e.g.:

```go
for _, tag := range tags {
// ...
}
```

👨‍🔬 https://go.dev/play/p/2AMPlzzzhkF

I would also run `WithRegistryAuth` once, outside of the `for` loop.

Final Go recipe (with a few more changes):

```go
package main

import (
    "context"
    "fmt"
    "os"

    "dagger.io/dagger"
)

func main() {
    // create Dagger client
    ctx := context.Background()
    client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
    if err != nil {
        panic(err)
    }
    defer client.Close()

    // load registry credentials from environment variables
    username := os.Getenv("DOCKERHUB_USERNAME")
    if username == "" {
      panic("DOCKERHUB_USERNAME env var must be set")
    }
    passwordPlaintext := os.Getenv("DOCKERHUB_PASSWORD")
    if passwordPlaintext == "" {
      panic("DOCKERHUB_PASSWORD env var must be set")
    }
    password := client.SetSecret("password", passwordPlaintext)

    // define multiple image tags
    tags := [4]string{"latest", "1.0-alpine", "1.0", "1.0.0"}

    // create and publish image with multiple tags
    ctr := client.Container().
        From("alpine").
        WithRegistryAuth("docker.io", username, password)

  for _, tag := range tags {
        addr, err := ctr.Publish(ctx, fmt.Sprintf("%s/alpine:%s", username, tag))
        if err != nil {
            panic(err)
        }
        fmt.Println("Published: ", addr)
    }
}
```

FWIW:
![image](https://user-images.githubusercontent.com/3342/236239824-bb016548-c514-42b5-b88f-6b9ce599c5c4.png)

Reply:
Thanks for the revision. Updated and modified other language snippets to match.